submissions/angels_unit_tests/angels_unit_tests.md

```diff
@@ -0,0 +1,55 @@
+---
+author: Lovely_Santa
+image:
+  url: none
+  transcription: A succesfull attempt at automated mod testing
+title: "Alt-F4 #999 - A succesfull attempt at automated mod testing..."
+lang: en
+discuss:
+  discord: https://alt-f4.blog/discord
+---
+
+After a long while, an old contributer *Lovely_Santa* is back in his writing pen! It has been since Issue #2 that we've heared from him, when he was experimenting with the modding abilities the (at the time) new spidertron. This time he stays on topic with the Angel's mods, while writing about something completely different...
+
+## How it all began...
+
+Before I tell you a story about automating the boring parts of creating a Factorio mod, I have to take you back to the beginning... Once upon a time, -no.. I have to start somewhere at the end of 2019. And by that I do not mean the well-known story of how COVID-19 started, but with the sad story about Arch666Angel stepping back from creating his all-famous [Angel's mods](https://mods.factorio.com/user/Arch666Angel). This all came due to IRL changes in his life. I'm not gonna bother explaining all the details, but it resulted in a few modders being interested in taking over his legacy.
+
+It took until the 27th of January 2020 to convince Arch666Angel to let the community contribute to his creation. After working out all the details, he selected few people to keep his vision of the mods alive, at which point we could inform the public about this. The main interaction at the time went through the factorio forums, from which you can still read the [announcement message](https://forums.factorio.com/viewtopic.php?p=475786#p475786) till this day.
+
+During the month January 2020 we had already eagerly started to update the mods from 0.17 to 0.18 with all the breaking changes the base game introduced. After Arch666Angel's approval, we released our first version [*"Initial 0.18 release"*](https://github.com/Arch666Angel/mods/commit/8b7ff10ba82d3071f374e08452b5ca22d42e4780)! Soon after celebrating our succes, reality slapped us in the face with a great amount of bug reports on the forums. Little did we know how huge this undertaking would be, how much time we spend bugfixing and most of all, how great the big community is by trying to help figure out the issues.
+
+## It started small
+
+With the great help of the community we learned how the Angel's libary worked. You can compare this to some stand-alone mods like the [Factorio Library](https://mods.factorio.com/mod/flib), [Bob's Functions Library mod](https://mods.factorio.com/mod/boblibrary) or even my own attempt [Lovely_santa's library of knowledge](https://mods.factorio.com/mod/LSlib). There is no 'right' or 'wrong' way to implement a library, as long as it suits its purpose. By understanding Angel's library, which actually lives inside Angel's refining and not as a stand-alone mod, we figured more efficient ways of dealing with certain bugs arrising in the mod.
```

Comment:
I think you should explain a little more what a `library` is, since it might not be aparent for non-programmers. Like that it's the foundation that the mods are built on, or functions that they all use. Could even add an example.

Reply:
Fixed in https://github.com/AlternativeFFFF/Alt-F4/pull/551/commits/0d2bc1d8be83f76d557f22d7d39ba53b96e0aa9f